web/client/components/mediaEditor/map/MapList.jsx

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2019, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React from "react";
+import {compose, mapPropsStream} from 'recompose';
+
+import MapCatalogComp from '../../maps/MapCatalog';
+import mapCatalog from '../../maps/enhancers/mapCatalog';
+import handleSelectEnhancer from '../../widgets/builder/wizard/map/enhancers/handleSelect';
+import Filter from '../../misc/Filter';
+import withLocal from "../../misc/enhancers/localizedProps";
+import SideGrid from '../../misc/cardgrids/SideGrid';
+import { SourceTypes, filterResources } from '../../../utils/GeoStoryUtils';
+import withFilter from '../../misc/enhancers/withFilter';
+
+const Icon = require('../../misc/FitIcon');
+
+const FilterLocalized = withLocal('filterPlaceholder')(Filter);
+
+const MapCatalog = compose(
+    mapCatalog,
+    /* the next stream updates the media Editor state, of maps in this case,
+     * every time the items change
+    */
+    mapPropsStream( props$ =>
+        props$.merge(
+            props$
+                .distinctUntilKeyChanged('items')
+                .do(({items = [], loadItems, selectedService, mediaType} = {}) => {
+                    if (items.length) {
+                        loadItems({
+                            mediaType,
+                            sourceId: selectedService,
+                            params: {mediaType},
+                            resultData: {
+                                resources: items.map(i => ({
+                                    ...i.map,
+                                    thumbnail: decodeURIComponent(i.map.thumbnail || "")
+                                }))
+                            }
+                        });
+                    }
+                })
+                .ignoreElements() // don't want to emit props
+    )))(MapCatalogComp);
+const defaultPreview = <Icon glyph="geoserver" padding={20} />;
+
+
+const MapList = handleSelectEnhancer(
```

Comment:
handleSelectEnhancer makes call onMapSelected this way
```
onMapSelected({map})
```
While you need to call updateItem(item);

add here an enhancer that transforms the handler (that is an adapter) instead of modelling all the other actions to match existing code.
Something like 
```javascript
withHandlers ({
    onMapSelected: ({onMapSelected = () => {}}) => ({map}) {
    // transform into a standard mediaEditor resource, if you need (I don't remember)
    onMapSelected(map);
};
})
```

Reply:
mmh, i think handle select should have generic names
when using it we should add an adapter that renames specific names into the generic one.
would this be good for you?

this would requrie a change to the components that are using this enhancer 
two actually: MapSelector (Widgets) and MapList (MediaEditor)